lib/order.rb

```diff
@@ -1,18 +1,90 @@
+require 'pry'
+require 'csv'
+require 'awesome_print'
+
+
 module Grocery
+  # begin
+  # rescue Grocery::FindError > e
+  # end
+
+  # custom error for the Order.find class method
+  class FindError < ArgumentError
+    def initialize(msg="Error: Order has not been created yet")
+      super
+      # rescue
+    end
+  end
+
   class Order
     attr_reader :id, :products
+    @@all_orders = []
 
     def initialize(id, products)
       @id = id
       @products = products
     end
 
     def total
-      # TODO: implement total
+      subtotal = 0
+      @products.each do |product, price|
+        subtotal += price
+      end
+      total = subtotal + (subtotal * 0.075)
+      return total.round(2)
     end
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      products = {product_name => product_price}
+      products.each do |product, price|
+        if @products.include? product
+          return false
+        else
+          @products.merge!({product => price})
+          return true
+        end
+      end
+    end # Order#add_product
+
+    def self.list_all
+      return @@all_orders
     end
-  end
-end
+
+    def self.populate
+      if @@all_orders.length == 0
+        @@all_orders = all
+      else
```

Comment:
I like this strategy of storing the orders you've loaded from disk in `@@all_orders`. This is a technique known as memoization, and it's a form of [cacheing](https://en.wikipedia.org/wiki/Cache_(computing)). This can have big performance benefits if you're reading the list of orders often, but it can also put your program in some sticky situations.

For example, what if another program updates the list of orders? Then your program would need a way to detect the change and reload the file.

For this program we don't need these sorts of complex behaviors, so cacheing the data works just fine.